File: 2199-two-furthest-houses-with-different-colors/2199-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int n=colors.length;
        int max=0;
        for(int i=n-1;i>=1;i--){
            if(colors[i]!=colors[0]){
                int value=i-0;
                max=Math.max(value,max);
                break;
            }
        }
        for(int i=0;i<n-1;i++){
            if(colors[n-1]!=colors[i]){
                int value=n-1-i;
                max=Math.max(value,max);
                break;
            }
        }
        
            
            
        return max;
    }
}

File: 2199-two-furthest-houses-with-different-colors/2199-two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        diff=0
        for i in range(len(colors)):
            for j in range(i+1,len(colors)):
                if colors[i]!=colors[j]:
                    diffv=abs(i-j)
                    diff=max(diff,diffv)
        return diff

        